### PR TITLE
feat(linter): eslint-plugin-jest: prefer-called-with

### PR DESCRIPTION
**Pull request by eryue0220**
Rule Detail: [link](https://github.com/jest-community/eslint-plugin-jest/blob/main/src/rules/prefer-called-with.ts)

### CODE REVIEW

File: crates/oxc_linter/src/rules/jest/prefer_called_with.rs [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+}
+
+impl PreferCalledWith {
+    pub fn run<'a>(possible_jest_node: &PossibleJestNode<'a, '_>, ctx: &LintContext<'a>) {
+        let node = possible_jest_node.node;
+        let AstKind::CallExpression(call_expr) = node.kind() else {
+            return;
+        };
+
+        if !is_type_of_jest_fn_call(call_expr, possible_jest_node, ctx, &[JestFnKind::Expect]) {
```
Comment (mysteryven) [inline crates/oxc_linter/src/rules/jest/prefer_called_with.rs:64]: Feel like we can remove this check, `parse_expect_jest_fn_call` will make sure it.

File: crates/oxc_linter/src/snapshots/prefer_called_with.snap [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+source: crates/oxc_linter/src/tester.rs
+assertion_line: 150
+expression: prefer_called_with
+---
+  ⚠ eslint-plugin-jest(prefer-called-with): Suggest using `toBeCalledWith()` or `toHaveBeenCalledWith()`.
+   ╭─[prefer_called_with.tsx:1:1]
+ 1 │ expect(fn).toBeCalled();
+   ·            ──────────
+   ╰────
+  help: Prefer "toBeCalled"With(/* expected args */)
```
Comment (mysteryven) [inline crates/oxc_linter/src/snapshots/prefer_called_with.snap:11]: Is there missing a space between `"toBeCalled"` and `with`?

**Review by mysteryven — CHANGES_REQUESTED** [review]

File: crates/oxc_linter/src/rules/jest/prefer_called_with.rs (hunk 1 above)
Comment (eryue0220) [reply crates/oxc_linter/src/rules/jest/prefer_called_with.rs]: Done~

File: crates/oxc_linter/src/snapshots/prefer_called_with.snap (hunk 2 above)
Comment (eryue0220) [reply crates/oxc_linter/src/snapshots/prefer_called_with.snap]: Update. The problem is double quote shouldn't be here. It should be displayed with `toBeCallWith(/* expected args*/)` or `toHaveBeenCallWith(/* expected args*/)`.

**Review by mysteryven — APPROVED** [review]
Thank you!!

**Pull request merged**